src/nft/components/sell/modal/ListingSection.tsx

```diff
@@ -0,0 +1,188 @@
+import clsx from 'clsx'
+import { Box } from 'nft/components/Box'
+import { Column, Row } from 'nft/components/Flex'
+import { ApprovedCheckmarkIcon, ChevronUpIcon, FailedListingIcon, LoadingIcon } from 'nft/components/icons'
+import { badge, bodySmall, buttonTextSmall, subhead } from 'nft/css/common.css'
+import { useSellAsset } from 'nft/hooks'
+import { AssetRow, CollectionRow, ListingRow, ListingStatus } from 'nft/types'
+import { formatEthPrice, numberToWei } from 'nft/utils/currency'
+import { useEffect, useState } from 'react'
+
+import * as styles from './ListingModal.css'
+
+export const ListingSection = ({
+  sectionTitle,
+  caption = undefined,
+  title = undefined,
+  rows,
+  index,
+  openIndex,
+  isSuccessScreen = false,
+}: {
+  sectionTitle: string
+  caption?: string
+  title?: string
+  rows: AssetRow[]
+  index: number
+  openIndex: number
+  isSuccessScreen?: boolean
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const notAllApproved = rows.some((row: AssetRow) => row.status !== ListingStatus.APPROVED)
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const removeAssetMarketplace = useSellAsset((state) => state.removeAssetMarketplace)
+
+  const removeRow = (row: any) => {
+    // collections
+    if (index === 1) {
+      for (const asset of sellAssets)
+        if (asset.asset_contract.address === row.collectionAddress) removeAssetMarketplace(asset, row.marketplace)
+    }
+    // listings
+    else removeAssetMarketplace(row.asset, row.marketplace)
+  }
+
+  useEffect(() => {
+    setIsOpen(index === openIndex ? true : false)
```

Comment:
I also feel like this can be passed as a boolean and doesn't need a useEffect. You can make a prop called `isOpen` and set it equal to `index === openIndex`

Reply:
The useEffect is used to automatically open up the subsequent when the previous section is completed, (ie all collections approved, open the signing section). We also need to be able to manually open and close the section with a button press.